src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8703,7 +8704,7 @@ private TypeWithState VisitConversion(
             bool canConvertNestedNullability = true;
             bool isSuppressed = false;
 
-            if (conversionOperand.IsSuppressed == true)
+            if (suppressed || conversionOperand.IsSuppressed)
```

Comment:
Would it be possible to unify the `suppressed` parameter with the `isSuppressed` local?